lib/cli/src/commands/init.rs

```diff
@@ -0,0 +1,549 @@
+            (_, true, true) => Err(anyhow::anyhow!(
+                "cannot initialize a wapm manifest with both --bin and --lib, pick one"
+            )),
+            (false, true, _) => Ok(BinOrLib::Bin),
+            (false, _, true) => Ok(BinOrLib::Lib),
+            _ => Ok(BinOrLib::Bin),
+        }
+    }
+
+    fn get_bindings(target_file: &Path, bin_or_lib: BinOrLib) -> Option<wapm_toml::Bindings> {
+        match bin_or_lib {
+            BinOrLib::Bin | BinOrLib::Empty => None,
+            BinOrLib::Lib => target_file.parent().and_then(|parent| {
+                walkdir::WalkDir::new(parent)
+                    .min_depth(1)
+                    .max_depth(3)
+                    .follow_links(false)
+                    .into_iter()
+                    .filter_map(|e| e.ok())
+                    .filter_map(|e| {
+                        let is_wit = e.path().extension().and_then(|s| s.to_str()) == Some(".wit");
+                        let is_wai = e.path().extension().and_then(|s| s.to_str()) == Some(".wai");
+                        if is_wit {
+                            Some(wapm_toml::Bindings::Wit(wapm_toml::WitBindings {
+                                wit_exports: e.path().to_path_buf(),
+                                wit_bindgen: semver::Version::parse("0.1.0").unwrap(),
+                            }))
+                        } else if is_wai {
+                            Some(wapm_toml::Bindings::Wai(wapm_toml::WaiBindings {
+                                exports: None,
+                                imports: vec![e.path().to_path_buf()],
+                                wai_version: semver::Version::parse("0.2.0").unwrap(),
+                            }))
+                        } else {
+                            None
+                        }
+                    })
+                    .next()
+            }),
+        }
+    }
+}
+
+struct ConstructManifestReturn {
+    namespace: String,
+    default_abi: wapm_toml::Abi,
+    module_name: String,
+    bindings: Option<wapm_toml::Bindings>,
+    toml: wapm_toml::Manifest,
+}
+
+fn construct_manifest(
+    cargo_toml: Option<&MiniCargoTomlPackage>,
+    package_name: &String,
+    target_file: &PathBuf,
+    manifest_path: &PathBuf,
+    bin_or_lib: BinOrLib,
+    namespace: Option<String>,
+    version: Option<semver::Version>,
+    template: Option<&Template>,
+    include_fs: &[String],
+) -> ConstructManifestReturn {
+    let package_name = cargo_toml.as_ref().map(|p| &p.name).unwrap_or(package_name);
+
+    let namespace = namespace.clone().unwrap_or_else(|| {
+        let username = wasmer_registry::whoami(None, None).ok().map(|o| o.1);
+        username
+            .or_else(|| package_name.split('/').next().map(|s| s.to_string()))
+            .unwrap_or_else(|| "_".to_string())
+    });
+    let module_name = package_name
+        .split('/')
+        .last()
+        .unwrap_or(package_name)
+        .to_string();
```

Comment:
I think [my previous comment](https://github.com/wasmerio/wasmer/pull/3324#discussion_r1047083376) got automatically marked as outdated when we pulled `construct_manifest()` out into its own function.

> This feels like a code smell.
> 
> If the package_name came from the cargo_toml variable then it can never contain a /. Similarly, if you scroll up about 100 lines you'll see that package_name is constructed using just the file_stem() of the current directory. That means the split() will always return the entire package_name string and we don't need to deal with any Options.
> 
> The deeper problem is that we shouldn't be using string.split("/") when dealing with package names because package names are more nuanced than just a string. Instead, we should try to use a high-level type and do something like string.parse::<PackageName>() so we can access its namespace.

Reply:
I removed this check, now the package name = the module name